### ISSUE TITLE
Finish Bundling Core Packages

### ISSUE THREAD

**Issue by confused-Techie** [pulsar-edit/pulsar#512]
### Thanks in advance for your bug report!

- [X] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [X] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [X] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [X] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [X] Have you checked you are on the latest release of Pulsar?

### What happened?

A goal of the Pulsar repo, for some time now has been to bundle any packages that are core packages. That is they are included in the final output of any installation of Pulsar.

The reason being, is bundling core packages allows us to archive the repos themselves, meaning we don't have to split our focus between so many codebases, and ensures any PRs to these core packages are able to get much more focus as this is by far the most popular repo. 

With all of that said, nearly all core packages are now bundled, with a few stragglers, that have posed problematic. Here I'll go ahead and consolidate these last items, as well as update their status, and hopefully ensure other contributors can see their path to be aware of the eventual goal of these items.

* [ ] `snippets`: Per core maintainers requests, this item will not be bundled until later notice 
* [ ] `autocomplete-atom-api`: #476 
* [ ] `symbols-view`: Per core maintainers requests, this item will not be bundled until the new tree-sitter changes land.
* [ ] `find-and-replace`: Has to many open PRs to bundle
* [ ] `fuzzy-finder`: Has to many open PRs to bundle
* [ ] `github`: Has to many open PRs to bundle
* [ ] `notifications`: Previously attempted on #424 and encountered a significantly increased test failure rate 
* [ ] `spell-check`: Has not been attempted or investigated
* [ ] `tree-view`: Has minimal open PRs that can be addressed prior to bundling

### Pulsar version

N/A

### Which OS does this happen on?

❓ Other(Please specify in the OS details field below)

### OS details

N/A

### Which CPU architecture are you running this on?

None

### What steps are needed to reproduce this?

N/A

### Additional Information:

N/A

**Comment by Sertonix** [comment]
While bundling the core packages some pull requests get lost. It would be nice if they would get a bit of attention.
I haven't checked all bundled packages but already mentioned a few lost pull requests.

**Comment by confused-Techie** [comment]
Yeah seems that has happened. Effort to bundle core packages has been shared recently, and I was not aware any were missing.

More than happy to ensure everything is getting bundled as properly as possible.

**Comment by savetheclocktower** [comment]
I want to offer a devil's advocate argument _against_ bringing `github` into the repo, and for a future where at least a couple of core packages live in their own repo:

I think most of the core packages are simple enough that it makes little sense to subject them to the rigors of arm’s-length development. It definitely makes it easier to fix small bugs if you don't have to open one PR, get it landed, then open another PR to bump the consumed version in Pulsar.

But I'd argue the more complex the package is, the more that isolation is justified.

The `github` package is _by far_ the most complex of all the core packages. In terms of lines of code, it might even rival Pulsar itself. It's got its own build scripts, its own babel config, its own autogenerated files. If all that stuff lived in the main Pulsar repo, it would be annoying in about five different ways.

Let me make this less of a “_feels_ like” argument and outline some traits that I think would warrant a package staying out of the repo, even if it's a core package:

1. Tool chain uniqueness. The `github` package is already unique in several ways. It's the only core package that uses React. If we ever get around to rewriting it like we want, it'll probably be in SolidJS or something, but it'd still be the only package that used SolidJS. I think it makes sense to hide the implementation details of the package in cases like this. It also frees up the package to be its own thing, rather than having to conform to the main repo's code style (such as it is).
2. Total weight. How many files are we talking about, and how many lines of code? How long would the test suite take to run? (We could use a couple of official packages as testers of `action-pulsar-dependency`.) The more the package actually does, the more it benefits from having its own bug tracker and CI pipeline.
3. I really wanted to think of a third thing, but it's not coming to mind right now. Maybe I'll edit it in later.

This is in my mind because I've been writing the [`terminal`](https://github.com/pulsar-edit/terminal) package in TypeScript. (I wanted type safety for several reasons, among them my lack of experience with XTerm.js.) That means a build process, plus rituals around checking in generated files. It means `cd`ing into a subdirectory of the project to run a specific `npm` task for development — but only for that one package. 

Consensus seems to be present to bring that package into core in the near future, but it'd be the first core package whose source is TypeScript. If it actually lived in the main repo, some contributors could forget (or fail to notice) that they should be editing the `.ts` files when they open PRs, and instead modify the built JS. I don't think it's a bad idea to have it live in its own house down the road from the core-package dormitory.

I don't think I'm making my best pitch here, and I could possibly be convinced otherwise. But if it were up to me, the `terminal` package would get added as an “external” core package just like `github`.
